lib/curl_req.ex

```diff
@@ -28,8 +28,24 @@ defmodule CurlReq do
     req
   end
 
-  defp run_steps(req) do
-    Enum.reduce(req.request_steps, req, fn {step_name, step}, req ->
+  defp run_steps(req, option) do
+    req.request_steps
+    |> Enum.filter(fn {step_name, _} ->
+      case option do
+        _ when is_boolean(option) ->
```

Comment:
This part is a little convoluted in my opinion. I'd like to at least see a pattern match on `false` to remove that branch from this path.

`def run_steps(req, false), do: req`